mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -369,6 +374,24 @@ impl Drop for RingBuffer {
     }
 }
 
+impl StreamWakeableState for RingBuffer {
+    fn insert(&mut self, value: &Publication) -> PersistResult<Key> {
+        RingBuffer::insert(self, value)
```

Comment:
What is the benefit of **not** putting the code inside implementation?

Reply:
None, just seemed easier than shifting entire fn contents over. But can do that.

Reply:
Moved everything into trait impl now :)